File: cpp/04_003.cpp
#include <iostream>

class Rectangle {
protected:
    int width;
    int height;
 
public:
    Rectangle() : width(0), height(0) {}
    
    virtual void setWidth(int w) {
        width = w;
    }
 
    virtual void setHeight(int h) {
        height = h;
    }
 
    int getArea() {
        return width * height;
    }
};
 
class Square : public Rectangle {
public:
    void setWidth(int w) override {
        width = w;
        height = w;
    }
 
    void setHeight(int h) override {
        height = h;
        width = h;
    }
};
 
int main() {
    Rectangle rectangle;
    rectangle.setWidth(5);
    rectangle.setHeight(10);
    int rectangleArea = rectangle.getArea();
    std::cout << "Area of rectangle: " << rectangleArea 
             << std::endl;
 
    Square square;
    square.setWidth(5);
    int squareArea = square.getArea();
    std::cout << "Area of square: " << squareArea << std::endl;
    
    return 0;
}

File: data structures and algorithms/python/04_003.py
class Rectangle:
    def __init__(self):
        self.width = 0
        self.height = 0
    
    def set_width(self, width):
        self.width = width
    
    def set_height(self, height):
        self.height = height
    
    def get_area(self):
        return self.width * self.height

class Square(Rectangle):
    def set_width(self, width):
        self.width = width
        self.height = width
    
    def set_height(self, height):
        self.height = height
        self.width = height

def main():
    rectangle = Rectangle()
    rectangle.set_width(5)
    rectangle.set_height(10)
    rectangle_area = rectangle.get_area()
    print(f"Area of rectangle: {rectangle_area}")
    
    square = Square()
    square.set_width(5)
    square_area = square.get_area()
    print(f"Area of square: {square_area}")

if __name__ == "__main__":
    main()